website/src/views/conceptMap.js

```diff
@@ -1,37 +1,45 @@
-import React from 'react';
-import { useContext, useEffect, useState } from 'react';
+import {useContext, useEffect, useState} from 'react';
 
 import api from '../utils/api';
 import Loader from '../components/Loader';
 import PageHeader from '../components/PageHeader';
 import './css/conceptMap.css';
-import { StudentSelectionContext } from "../components/StudentSelectionWrapper";
+import jwtDecode from "jwt-decode";
+import apiv2 from "../utils/apiv2";
+import {StudentSelectionContext} from "../components/StudentSelectionWrapper";
 
 export default function ConceptMap() {
     const [loading, setLoading] = useState(false);
     const [studentMastery, setStudentMastery] = useState('000000');
-    const { selectedStudent, getSelectedStudent } = useContext(StudentSelectionContext);
+
+    const {selectedStudent, setSelectedStudent} = useContext(StudentSelectionContext);
 
     useEffect(() => {
         let mounted = true;
         setLoading(true);
         if (mounted && localStorage.getItem('token')) {
-            if (selectedStudent === "") {
-                api.get('/progressquerystring').then((res) => {
-                    setStudentMastery(res.data);
-                    setLoading(false);
-                });
-            } else {
-                api.get('/admin/studentProgressReport?email=' + selectedStudent).then((res) => {
-                    setStudentMastery(res.data);
-                    setLoading(false);
-                });
-            }
+            let email = jwtDecode(localStorage.getItem('token')).email;
+            api.get('v2/students/' + email + '/progressquerystring').then((res) => {
+                setStudentMastery(res.data)
+                setLoading(false);
+            });
         } else {
             setLoading(false);
         }
         return () => mounted = false;
-    }, [selectedStudent]);
+    }, []);
```

Comment:
We should prob leave the bound in if we can.  Avoid making unnecessary updates.

Reply:
selectedStudent is the current value of the student navbar dropdown, so it does update when the user changes the selected student